### PR TITLE
Match FullWithdrawal requests with forced actions in state update

### PR DESCRIPTION
**Pull request by adamiak**
Resolves L2B-790

Match FullWithdrawal requests with forced actions from SpotCairoOutput.modifications

### CODE REVIEW

File: packages/backend/src/core/StateUpdater.test.ts [hunk 1]
```diff
@@ -261,8 +275,82 @@ describe(StateUpdater.name, () => {
...
+            starkKey: StarkKey.fake('aaa'),
+            vaultId: 45n,
+          },
+        },
+      ]
+
+      const userTransactionRepository = mock<UserTransactionRepository>({
+        async getNotIncluded() {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          return transactions as UserTransactionRecord<any>[]
```
[the remark this answers is not in the document]
Comment (adamiak) [reply packages/backend/src/core/StateUpdater.test.ts:322]: Good question.

I've copied the test structure from the previous tests in this file. I'm not sure why "any" is necessary here - I tried to refactor it out but there's some high level typescript magic going on and I'm not sure how to resolve it. Given that other test in the file already does things this way, I'd prefer to keep it and not spend more time on it.

File: packages/backend/src/core/collectors/UserTransactionCollector.test.ts [hunk 2]
```diff
@@ -150,6 +151,63 @@ describe(UserTransactionCollector.name, () => {
...
+  it(`can process ${LogFullWithdrawalRequest.name}`, async () => {
+    const blockRange = new BlockRange([], 100, 200)
+    const starkKey = StarkKey.fake('123')
+    const vaultId = 123n
+    const transactionHash = Hash256.fake('abc')
+    const starkExAddress = EthereumAddress.fake('def')
+
+    const ethereumClient = mock<EthereumClient>({
+      async getLogsInRange(range, parameters) {
+        expect(range).toEqual(blockRange)
```
[the remark this answers is not in the document]
Comment (adamiak) [reply packages/backend/src/core/collectors/UserTransactionCollector.test.ts:163]: Personally I don't like it but that's how all other events are tested in this file. Let's keep if for now and discuss separately if this whole file should be refactored.

File: packages/backend/src/peripherals/database/transactions/UserTransaction.test.ts [hunk 3]
```diff
@@ -30,6 +31,25 @@ describe(encodeUserTransactionData.name, () => {
...
+      type: 'FullWithdrawal',
+      starkKey: StarkKey.fake(),
+      vaultId: 1234n,
+    }
+    const encoded = encodeUserTransactionData(data)
+
+    expect(encoded).toEqual({
+      starkKeyA: data.starkKey,
+      vaultOrPositionIdA: data.vaultId,
+      data: expect.anything(),
```
[the remark this answers is not in the document]
Comment (adamiak) [reply packages/backend/src/peripherals/database/transactions/UserTransaction.test.ts]: Fixed.

File: packages/backend/src/core/StateUpdater.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wendrowiec13) [reply packages/backend/src/core/StateUpdater.test.ts:322]: Sure

File: packages/backend/src/core/collectors/UserTransactionCollector.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wendrowiec13) [reply packages/backend/src/core/collectors/UserTransactionCollector.test.ts:163]: Sure

File: packages/backend/src/core/StateUpdater.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export class StateUpdater<T extends PositionLeaf | VaultLeaf> {
   async processStateTransition(
     stateTransitionRecord: StateTransitionRecord,
     expectedPositionRoot: PedersenHash,
-    forcedActions: ForcedAction[],
+    forcedActions: (ForcedAction | SpotModification)[],
```
[the remark this answers is not in the document]
Comment (adamiak) [reply packages/backend/src/core/StateUpdater.ts]: Done, refactored

**Review by sz-piotr — APPROVED** [review]

**Pull request merged**